tests/checkpoint/test_v1_checkpoint.py

```diff
@@ -329,3 +339,275 @@ def test_checkpoint_deserialization_failure(
+
+    @pytest.mark.unit
+    def test_checkpoint_run_with_actions(self):
+        pass  # TBD
+
+    @pytest.mark.unit
+    def test_result_init_no_run_results_raises_error(self, mocker: pytest.MockFixture):
+        with pytest.raises(ValueError) as e:
+            CheckpointResult(
+                run_id=mocker.Mock(spec=RunIdentifier),
+                run_results={},
+                checkpoint_config=mocker.Mock(spec=Checkpoint),
+            )
+
+        assert "CheckpointResult must contain at least one run result" in str(e.value)
+
+    @pytest.mark.unit
+    def test_result_describe(self, mocker: pytest.MockFixture):
+        result = CheckpointResult(
+            run_id=mocker.Mock(spec=RunIdentifier),
+            run_results={
+                mocker.Mock(spec=ValidationResultIdentifier): mocker.Mock(
+                    spec=ExpectationSuiteValidationResult, success=True
+                ),
+            },
+            checkpoint_config=mocker.Mock(spec=Checkpoint),
+        )
+
+        with mock.patch.object(
+            ExpectationSuiteValidationResult,
+            "describe_dict",
+            return_value={
+                "success": True,
+                "statistics": {
+                    "evaluated_expectations": 1,
+                    "successful_expectations": 1,
+                    "unsuccessful_expectations": 0,
+                    "success_percent": 100.0,
+                },
+                "expectations": [
+                    {
+                        "expectation_type": "expect_column_values_to_be_between",
+                        "success": True,
+                        "kwargs": {
+                            "batch_id": "default_pandas_datasource-#ephemeral_pandas_asset",
+                            "mostly": 0.95,
+                            "column": "passenger_count",
+                            "min_value": 0.0,
+                            "max_value": 6.0,
+                        },
+                        "result": {
+                            "element_count": 100000,
+                            "unexpected_count": 1,
+                            "unexpected_percent": 0.001,
+                            "partial_unexpected_list": [7.0],
+                            "unexpected_percent_total": 0.001,
+                            "unexpected_percent_nonmissing": 0.001,
+                            "partial_unexpected_counts": [{"value": 7.0, "count": 1}],
+                            "partial_unexpected_index_list": [48422],
+                        },
+                    },
+                ],
+            },
+        ):
+            actual = result.describe_dict()
+
+        validation_results = actual.pop("validation_results")
+        assert len(validation_results) == 1
+
+        expected = {"success": True, "success_percent": 100.0}
+        assert actual == expected
+
+    @pytest.mark.filesystem
+    def test_checkpoint_run_filesytem_e2e(self, tmp_path: pathlib.Path):
```

Comment:
Since we're getting into precarious territory with upcoming deletions of legacy code, I wanted extra coverage here. The unit tests should also cover these code paths but I wanted to showcase how a user might actually run this domain object in production